var/spack/repos/builtin/packages/otf2/package.py

```diff
@@ -27,6 +27,10 @@ class Otf2(AutotoolsPackage):
     def configure_args(self):
         return [
             '--enable-shared',
+            'CC={0}'.format(self.compiler.cc),
+            'CXX={0}'.format(self.compiler.cxx),
+            'F77={0}'.format(self.compiler.f77),
+            'FC={0}'.format(self.compiler.fc),
```

Comment:
I think this will  cause OTF2 not to be RPATH'd properly.  I'm assuming you're doing this so that *its* compiler wrappers use the "real" cc/cxx/f77/fc, right?  To get that, but also have OTF2 built with Spack's compiler wrappers, look at MPICH here:

https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/mpich/package.py#L70

We use the `filter_compiler_wrappers()` directive to patch the wrappers *after* they are installed.  We do this for `mpich`,  `openmpi`, and other packages that have scripts that need to refer to compilers they were built with.

Does that help?

Reply:
Thanks for your comment!
If `self.compiler.xxx` is set to `CC/CXX/F77/FC`, does it mean that the compiler's library cannot be referenced correctly from the BINARY? 

This time, the problem was that `otf2`'s `configure` did not recognize the each compile commands. So I think that the arguments passed to configure can be the "real" compiler commands. I fixed it on commit of "Modify the reference of the compile commands.". Could you confirm the correction?